File: 프로그래머스 level 2/오픈채팅방.py
def solution(record):
    answer = list()
    table = {}
    for data in record:
        if data.split()[0] == 'Enter' or data.split()[0] == 'Change':
            table[data.split()[1]] = data.split()[2]

    for data in record:
        if data.split()[0] == 'Enter':
            answer.append('{}님이 들어왔습니다.'.format(table[data.split()[1]]))
        elif data.split()[0] == 'Leave':
            answer.append('{}님이 나갔습니다.'.format(table[data.split()[1]]))

    return answer

File: Swift/프로그래머스 level 2/오픈채팅방.swift
import Foundation

func solution(_ recordList:[String]) -> [String] {
    var nicknameDict: Dictionary = [String: String]()
    
    var answerList : [String] = []
    
    for recordSet in recordList {
        let record = recordSet.components(separatedBy: " ")
        if record[0] == "Enter" || record[0] == "Change" {
            nicknameDict[record[1]] = record[2]
        }
    }
    
    for recordSet in recordList {
        let record = recordSet.components(separatedBy: " ")
        
        // let userName = nick
        
        if record[0] == "Enter" {
            answerList.append("\(nicknameDict[String(record[1])] ?? "")님이 들어왔습니다.")
        } else if record[0] == "Leave" {
            answerList.append("\(nicknameDict[String(record[1])] ?? "")님이 나갔습니다.")
        }
    }
    
    return answerList
    
}